components/datasource-selector/layman/layman.service.ts

```diff
@@ -0,0 +1,126 @@
+import {HttpClient} from '@angular/common/http';
+import {Injectable} from '@angular/core';
+import {catchError} from 'rxjs/operators';
+
+import {HsEndpoint} from '../../../common/endpoints/endpoint.interface';
+import {HsUtilsService} from '../../utils/utils.service';
+
+@Injectable({providedIn: 'root'})
+export class HsLaymanBrowserService {
+  constructor(
+    private http: HttpClient,
+    private hsUtilsService: HsUtilsService
+  ) {
+    'ngInject';
+  }
+
+  /**
+   * @function queryCatalog
+   * @memberof HsLaymanBrowserService
+   * @param {object} endpoint Configuration of selected datasource (from app config)
+   * extent feature is created. Has one parameter: feature
+   * @description Loads datasets metadata from Layman
+   */
+  queryCatalog(endpoint: HsEndpoint): void {
+    endpoint.getCurrentUserIfNeeded(endpoint);
+    let url = `${endpoint.url}/rest/${endpoint.user}/layers`;
+    url = this.hsUtilsService.proxify(url);
+    endpoint.datasourcePaging.loaded = false;
+    /*if (endpoint.canceler !== undefined) {
```

Comment:
Some cancelation of metadata catalogue queries ill be needed Im afraid because when the queries return the results after a long time (say 20 secs) then debounce is not enough.